### PR TITLE
Separate out envBuilder for building environment before rollback

### PR DESCRIPTION
**Pull request by baixinsui**

### CODE REVIEW

File: plugins/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/HuaweiCloudOrchestratorPlugin.java [hunk 1]
```diff
@@ -76,15 +73,22 @@ public void startManagedService(String managedServiceName) {
...
         BuilderContext ctx = new BuilderContext();
         ctx.setConfig(config);
 
-        if (optionalAtomBuilder.isEmpty()) {
-            throw new IllegalStateException("Builder not found.");
+        BuilderFactory factory = new BuilderFactory();
+        Optional<AtomBuilder> optionalEnvBuilder = factory.createBuilder(
+            BuilderFactory.ENV_BUILDER, ocl);
+
+        if (optionalEnvBuilder.isEmpty()) {
```
Comment (iskey) [inline plugins/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/HuaweiCloudOrchestratorPlugin.java:83]: How about we raise this exception from `createBuilder` directly?

File: plugins/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/HuaweiCloudOrchestratorPlugin.java [hunk 2]
```diff
@@ -111,15 +117,28 @@ public void stopManagedService(String managedServiceName) {
...
+        }
         BuilderContext ctx = new BuilderContext();
         ctx.setConfig(config);
 
-        if (optionalAtomBuilder.isEmpty()) {
-            throw new IllegalStateException("Builder not found.");
+        BuilderFactory factory = new BuilderFactory();
+        Optional<AtomBuilder> optionalEnvBuilder =
+            factory.createBuilder(BuilderFactory.ENV_BUILDER, ocl);
+        if (optionalEnvBuilder.isEmpty()) {
```
Comment (iskey) [inline plugins/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/HuaweiCloudOrchestratorPlugin.java:131]: As mentioned above.

**Review by iskey — CHANGES_REQUESTED** [review]

File: plugins/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/HuaweiCloudOrchestratorPlugin.java (hunk 1 above)
Comment (baixinsui) [reply plugins/huaweicloud/src/main/java/org/eclipse/osc/orchestrator/plugin/huaweicloud/HuaweiCloudOrchestratorPlugin.java]: throw not supported exception in createBuilder

**Review by iskey — APPROVED** [review]

**Comment by iskey** [comment]
LGTM

**Pull request merged**